scripts/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -388,6 +404,62 @@ def run_spearman_correlation_scatter(
     # don't have an alt_allele value 
     gene_snp_df = gene_snp_df[gene_snp_df.snpid.isin(set(genotype_df.snpid))]
 
+    # Get association effect data, to be used for violin plots for each genotype of each SNP
+    def get_association_effect_data(gene):
+        sampleid = expression_df.sampleid
+        log_cpm = calculate_log_cpm(expression_df)
+        log_cpm['sampleid'] = sampleid
+        # reduce log_cpm matrix to gene of interest only
+        log_cpm = log_cpm[['sampleid', gene]]
+        # merge log_cpm data with genotype info 
+        gt_expr_data = genotype_df.merge(log_cpm, left_on='sampleid', right_on='sampleid')
+        # group gt_expr_data by snpid and genotype
+        grouped_gt = gt_expr_data.groupby(['snpid', 'n_alt_alleles'])
+
+        def create_struct(gene, group):
+            hist, bin_edges = np.histogram(group[gene], bins=10)
+            n_samples = group[gene].count()
+            min_val = group[gene].min()
+            max_val = group[gene].max()
+            mean_val = group[gene].mean()
+            q1, median_val, q3 = group[gene].quantile([0.25, 0.5, 0.75])
+            iqr = q3 - q1
+            # not sure what this value is doing here
```

Comment:
What is this comment referring to - if the interquartile ranges aren't used for the violin plots, do we need to calculate them and include them in the struct?

Reply:
Sorry, that's a very disconcerting comment to have in code 🤦‍♀️ As @daniaki said, that's a holdover from previously, when I wasn't sure what the calculation was used for, as I got it from the GitHub issues doc. What I found out is that that calculation is used to identify outliers (hence, the 1.5 constant). @daniaki no problem, so since the min and max value are already there, just keep delete the `iqr_min` and `iqr_max`? I'll go ahead and remove them from the code and the data struct